apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/SWExecutionHookWrapper.java

```diff
@@ -44,10 +49,15 @@ public SWExecutionHookWrapper(HystrixCommandExecutionHook actual) {
 
         EnhancedInstance enhancedInstance = (EnhancedInstance) commandInstance;
         EnhanceRequireObjectCache enhanceRequireObjectCache = (EnhanceRequireObjectCache) enhancedInstance.getSkyWalkingDynamicField();
-        if (ContextManager.isActive()) {
-            enhanceRequireObjectCache.setContextSnapshot(ContextManager.capture());
+        try {
+            if (ContextManager.isActive()) {
```

Comment:
@BFergerson isActive should have checked the TracingContext is there,  why NPE in the `#capture`?

@Indifer Your fix is not right, because this should not happen at all. We need to find out why. The reason is above.

Reply:
ContextManager.isActive() is judge AbstractTracerContext is not null。
![image](https://user-images.githubusercontent.com/7918822/79678912-e6c70600-8232-11ea-8b98-ea39b40018b0.png)

but ContextManager.capture() may be throw ex
![image](https://user-images.githubusercontent.com/7918822/79678915-efb7d780-8232-11ea-9f87-590b6760c0c4.png)
![image](https://user-images.githubusercontent.com/7918822/79678939-46251600-8233-11ea-92bf-9ed3ce08d997.png)

![image](https://user-images.githubusercontent.com/7918822/79678921-fc3c3000-8232-11ea-9974-91c386006377.png)

Reply:
With no span in the context, shouldn't have the TracingContext. That is my point.

Reply:
@wu-sheng, I'm not familiar with Hystrix through I agree this change isn't the solution. My first thought is maybe the version of Hystrix @Indifer is using isn't supported yet. I think we would need a reproducible example to move forward.

Reply:
Agree. @Indifer I hope you could debug more about why this happens.

Reply:
The version of Hystrix is 1.5.11, I'm not familiar with Hystrix too, I'll debug it sometime.